apps/api/src/app/workflows-v2/usecases/build-payload-schema/build-payload-schema.usecase.ts

```diff
@@ -87,4 +73,30 @@ export class BuildPayloadSchema {
 
     return processedValue;
   }
+
+  private async buildVariablesSchema(variables: string[]) {
+    // TODO: Update typings in this as .payload can be null or undefined
+    const variablesObject = pathsToObject(variables, {
+      valuePrefix: '{{',
+      valueSuffix: '}}',
+    }).payload;
+
+    const schema: JSONSchemaDto = {
+      type: 'object',
+      properties: {},
+      required: [],
+      additionalProperties: true,
+    };
+
+    if (variablesObject) {
+      for (const [key, value] of Object.entries(variablesObject)) {
```

Comment:
will this for loop cover nested payload as well, for example, `payload.bar.data.foo...` etc..?

Reply:
Note, this is the reason e2e is failing, and I am pushing a fix.